src/native/corehost/hostpolicy/hostpolicy_context.cpp

```diff
@@ -80,6 +85,19 @@ namespace
             return CryptoResolveDllImport(entrypointName);
         }
 #endif
+        // there are two PInvokes in the hostpolicy itself, redirect them here.
+        if (strcmp(libraryName, hostPolicyLib) == 0)
+        {
+            if (strcmp(entrypointName, "corehost_resolve_component_dependencies") == 0)
+            {
+                return (void*)corehost_resolve_component_dependencies;
+            }
+
+            if (strcmp(entrypointName, "corehost_set_error_writer") == 0)
+            {
+                return (void*)corehost_set_error_writer;
+            }
```

Comment:
It is ok for any user to try calling methods that do not exist in hostpolicy. It is also (although very unlikely) that unrelated hostpolicy.dll may exist and user PInvokes something from it that is real. Also there could be additional overrides registered at higher level, - in theory. 

Basically, while very unlikely, there could be scenarios when not finding an override for a hostpolicy PInvoke is expected and ok.

I think, as a matter of testing in controlled environment, an assert makes sense here, but not a warning, since, strictly speaking this is not a failure.

Reply:
Actually even an assert could become a blocking nuisance if someone needs to use Checked/Debug build and there is a call to something that does not exist in hostpolicy. 
Otherwise it should be a "method missing" exception that can be caught and handled/ignored.

I think this part should stay as-is.